### ISSUE TITLE
No "loading/saving" state in Modal

### ISSUE THREAD

**Issue by ethanwinters** [IBM/carbon-components-react#1636]
### Summary

In a modal when someone pushes save there is no pattern built into the Modal component to show a loading state (like the loading state in this PR https://github.com/IBM/carbon-components-react/pull/1012)

### Justification

We save many items from inside a modal, there is no nice way to provide feedback to a user that we are processing

### "Must have" functionality

1) Ability to let the modal know it should be in a "saving" state
2) Ability to pass (optional) text to accompany inline-loading
3) Ability to mark primary and secondary buttons as disabled while in "saving" state... or perhaps in a form invalid state (not primary issue, but seems reasonable)

### Available extra resources

We can dedicate resources to work on this issue with Carbon team approval

**Comment by mattrosno** [comment]
@aagonzales in the [inline loading usage](https://www.carbondesignsystem.com/components/inline-loading/usage) guidelines...

> This could be in a table, after a primary or secondary button click, or even in a modal.

But we don't say how. Is it inline with the existing buttons? Does it replace the primary button? Some other location in the modal? Also in the guidelines...

> If the Inline Loading is being used to submit a form, the form fields should get disabled.

If we add a loading state to the modal, do we use that to disable inputs while it's loading?

@asudoh including you since I see you have the most recent modal commits. Thanks!

**Comment by mattrosno** [comment]
[Input modal](http://react.carbondesignsystem.com/?selectedKind=ModalWrapper&selectedStory=input%20modal&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Fstories%2Fstories-panel) would be a good story to demonstrate a loading state.

**Comment by aagonzales** [comment]
Yeah its just one of the interaction we never built out. But I'm assuming it would be something similar to what Logan has in #1012 PR. Its not the best visual solution but we can work on improving that in V10. And yes, the fields will be disabled since in modals as well since that data is being submitted.  

![image](https://user-images.githubusercontent.com/11670886/49835321-7073e300-fd64-11e8-9182-dd39e28ed398.png)

**Comment by joshblack** [comment]
cc @IBM/carbon-designers

**Comment by dakahn** [comment]
@carbon-design-system/design ⚡️

**Comment by designertyler** [comment]
The team shared mockups earlier this week that looked like a solid application of the behavior for the v10 modal.

@ethanwinters could you post those visuals here? 

Our guidance is limited and doesn't cover the error state. The team had a mockup with an inline notification for the error message and it seemed like an ok solution, but something we could work on in our guidance.

**Comment by ethanwinters** [comment]
@designertyler I think we have patterns for form field errors, what the current component did not have was the ability to disable the buttons (something you would do if your form was in an error state)
